schema_registry.cc

```diff
@@ -274,39 +275,70 @@ global_schema_ptr::global_schema_ptr(global_schema_ptr&& o) noexcept {
     assert(o._cpu_of_origin == current);
     _ptr = std::move(o._ptr);
     _cpu_of_origin = current;
+    _base_schema = std::move(o._base_schema);
 }
 
 schema_ptr global_schema_ptr::get() const {
     if (this_shard_id() == _cpu_of_origin) {
         return _ptr;
     } else {
         // 'e' points to a foreign entry, but we know it won't be evicted
-        // because _ptr is preventing this.
-        const schema_registry_entry& e = *_ptr->registry_entry();
-        schema_ptr s = local_schema_registry().get_or_null(e.version());
-        if (!s) {
-            s = local_schema_registry().get_or_load(e.version(), [&e](table_schema_version) {
-                return e.frozen();
-            });
+        // because _ptr / _base_schema.
```

Comment:
I don't understand the new meaning of this comment - `but we know it won't be evicted because _ptr / _base_schema` seems to be an incomplete sentence. Could you elaborate on what happens here?

Reply:
I will change the comment to be more informative. The basic idea behind it is this - one can think that we have a race here. We have an `lw_shared_ptr` dereference (few lines below: `*_base_schema->registry_entry()`) but we 
dereference it from possibly another shard (the whole idea of `get()` is to return a localized schema as if the schema is actually global and can be used from any shard). The reason this specific case is not a race, is that we hold a pointer to the schema in the creating shard and this is valid as long as the global pointer object is alive, so if we got the global_pointer object we know that this dereference is guarantied to succeede.